src/gui/torrentcontenttreeview.cpp

```diff
@@ -106,6 +108,21 @@ void TorrentContentTreeView::keyPressEvent(QKeyEvent *event)
     }
 }
 
+void TorrentContentTreeView::mouseDoubleClickEvent(QMouseEvent *event)
```

Comment:
Is there no easier way than to manually track mouse clicks in a certain area?

Reply:
I might assume that a model or delegate might allow this to be done in a more convenient way.

Reply:
At least I would connect to `QAbstractItemView::doubleClicked()` signal, check index and emit `itemDoubleClicked()` if it is appropriate.

Reply:
> I might assume that a model or delegate might allow this to be done in a more convenient way.

Model handle data, delegate handles painting. the delegate doesn't receive mouse events.

> At least I would connect to QAbstractItemView::doubleClicked() signal, check index and emit itemDoubleClicked() if it is appropriate.

Then I have to do extra to get the mouse position.

Reply:
>Then I have to do extra to get the mouse position.

Sorry, I don't have any compuy on hand to look at it... Isn't checkbox in a separate column?

Reply:
> Isn't checkbox in a separate column?

Reply:
>>Isn't checkbox in a separate column?
>
>nope

Okay, then it looks like there's nothing better to do.

I wonder why not use a separate column for checkbox? (apparently, the question is rhetorical)

Reply:
> I wonder why not use a separate column for checkbox? 

IMO won't look good.